commands/src/main/java/org/wildfly/extras/creaper/commands/orb/Attribute.java

```diff
@@ -0,0 +1,82 @@
+package org.wildfly.extras.creaper.commands.orb;
+
+
+/**
+ * Definition of attribute transition state. As builder defines
+ * it could be
+ * <ul>
+ *   <li>having value - current value will be redefined</li>
+ *   <li>having no value (being null) - leaving current value where it is</li>
+ *   <li>undefined value - current value will be undefined</li>
+ * </ul>
+ */
+final class Attribute<T> {
+    private final T value;
+    private final boolean isToUndefine;
+
+    private Attribute(T value) {
+        this.value = value;
+        this.isToUndefine = false;
+    }
+
+    private Attribute(T value, boolean isToUndefine) {
+        this.value = value;
+        this.isToUndefine = isToUndefine;
+    }
+
+    public static <T> Attribute<T> noValue() {
+        return new Attribute<T>(null);
+    }
+
+    public static <T> Attribute<T> undefine() {
+        return new Attribute<T>(null, true);
+    }
+
+    public static <O> Attribute<O> undefine(O object) {
```

Comment:
it isn't used, I've removed it. It was left in code from pre-pull request dates where it was used in command builder.

my lack of knowledge. I was thinking about `Class<? extends T> clazz` but I got some trouble to do so which I can't remember. And I didn't know that is considered to be an idiomatic solution.
Btw. why this does not happen for Java 8 (for not stealing your time, some link would be enough :)?

btw. a longer time I'm wondering why `Arrays#toArray(T[] a)` method (https://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#toArray(T[])) is not defined in similar way as you mentioned?